crates/context_aware_config/src/api/context/types.rs

```diff
@@ -1,11 +1,61 @@
+use derive_more::{AsRef, Deref};
 use serde::{Deserialize, Serialize};
-use serde_json::{Map, Value};
+use serde_json::{json, Map, Value};
+use service_utils::{bad_argument, result as superposition};
+
+#[derive(Deserialize, Clone, AsRef, Deref, Debug, Eq, PartialEq)]
+#[serde(try_from = "Map<String,Value>")]
+pub struct Condition(Map<String, Value>);
+
+impl Condition {
+    pub fn new(condition_map: Map<String, Value>) -> superposition::Result<Self> {
+        if !condition_map.is_empty() {
+            jsonlogic::apply(&json!(condition_map), &json!({})).map_err(|msg| {
```

Comment:
We can use jsonlogic::compile here if we want to check validity instead of applying it